x/concentrated-liquidity/pool.go

```diff
@@ -340,3 +340,22 @@ func validateAuthorizedQuoteDenoms(ctx sdk.Context, denom1 string, authorizedQuo
 func (k Keeper) GetLinkedBalancerPoolID(ctx sdk.Context, concentratedPoolId uint64) (uint64, error) {
 	return k.gammKeeper.GetLinkedBalancerPoolID(ctx, concentratedPoolId)
 }
+
+// GetTotalLiquidity gets the total liquidity across all concentrated pools.
+func (k Keeper) GetTotalLiquidity(ctx sdk.Context) (sdk.Coins, error) {
+	coins := sdk.Coins{}
+	pools, err := k.GetPools(ctx)
+	if err != nil {
+		return nil, err
+	}
+	for _, poolI := range pools {
+		pool, ok := poolI.(types.ConcentratedPoolExtension)
+		if !ok {
+			return nil, fmt.Errorf("pool is not a concentrated pool")
+		}
+		poolAddress := pool.GetAddress()
+		poolLiquidity := k.bankKeeper.GetAllBalances(ctx, poolAddress)
+		coins = coins.Add(poolLiquidity...)
+	}
+	return coins, nil
+}
```

Comment:
Consider defining this on the querier so that this keeper method is not accidentally used in the state-machine

Reply:
This isn't straight forward to me how I would do this, the querier doesn't have access to the bank keeper. Unless you are suggestion something else that I didn't pick up.

Reply:
We can make the bank keeper a field of the querier while setting it in the module.

Also, I'm okay with not blocking the PR on this comment.

Reply:
I ended up adding GetTotalLiquidity to PoolI and so now each module has this query at the keeper level (this should be safe for use in state machine)